### PR TITLE
feat: [VAN-1000] - Set weekly goal through welcome email via query param

### PR DESCRIPTION
**Pull request by shafqatfarhan**
**Description**
Set weekly goal via query param and delete param to avoid setting the goal multiple times.

**Ticket**
https://2u-internal.atlassian.net/browse/VAN-1000

### CODE REVIEW

**Review by zainab-amir — APPROVED** [review]

File: src/course-home/outline-tab/widgets/WeeklyLearningGoalCard.jsx [hunk 1]
```diff
@@ -65,6 +69,21 @@ function WeeklyLearningGoalCard({
     }
   }
 
+  useEffect(() => {
+    const currentParams = new URLSearchParams(location.search);
+    const weeklyGoal = Number(currentParams.get('weekly_goal'));
+    if (weeklyGoal) {
+      handleSelect(weeklyGoal, true);
```
Comment (cdeery) [inline src/course-home/outline-tab/widgets/WeeklyLearningGoalCard.jsx:76]: I think the input needs to be checked for correct values before it is consumed.

File: src/course-home/outline-tab/widgets/WeeklyLearningGoalCard.jsx [hunk 2]
```diff
@@ -32,8 +34,9 @@ function WeeklyLearningGoalCard({
   const [daysPerWeekGoal, setDaysPerWeekGoal] = useState(daysPerWeek);
   // eslint-disable-next-line react/prop-types
   const [isGetReminderSelected, setGetReminderSelected] = useState(subscribedToReminders);
+  const location = useLocation();
 
-  function handleSelect(days) {
+  function handleSelect(days, triggeredFromEmail = false) {
     // Set the subscription button if this is the first time selecting a goal
     const selectReminders = daysPerWeekGoal === null ? true : isGetReminderSelected;
     setGetReminderSelected(selectReminders);
```
Comment (cdeery) [inline src/course-home/outline-tab/widgets/WeeklyLearningGoalCard.jsx:42]: Am I correct that the expectation here is that if the learner does not want to be subscribed to email reminders then they have the opportunity to do that when they go to the page? Does the focus need to be set to the unsubscribe button? How would users with assistive technology know they could unsubscribe?

**Review by cdeery — CHANGES_REQUESTED** [review]
A couple of comments

File: src/course-home/outline-tab/widgets/WeeklyLearningGoalCard.jsx (hunk 1 above)
Comment (shafqatfarhan) [reply src/course-home/outline-tab/widgets/WeeklyLearningGoalCard.jsx:76]: done

File: src/course-home/outline-tab/widgets/WeeklyLearningGoalCard.jsx (hunk 2 above)
Comment (shafqatfarhan) [reply src/course-home/outline-tab/widgets/WeeklyLearningGoalCard.jsx:42]: Yes, learners have an option to unsubscribe to reminder emails. Currently, the focus is not set to the unsubscribe button even when the user manually sets the goal from the UI.

To support assistive technology, we are already setting the `aria-checked` attribute to indicate the currently selected weekly goal button in their group.

**Comment by shafqatfarhan** [comment]
@cdeery PR is ready for review again. Thanks.

**Review by cdeery — APPROVED** [review]
Thanks for the changes. Looks good

**Pull request merged**